src/Estimators/src/FloatingBaseEstimator.cpp

```diff
@@ -604,3 +637,73 @@ bool FloatingBaseEstimator::setupFixedVectorParamPrivate(const std::string& para
     return true;
 }
 
+bool FloatingBaseEstimator::updateBaseStateFromIMUState(const FloatingBaseEstimators::InternalState& state,
+                                                        const FloatingBaseEstimators::Measurements& meas,
+                                                        iDynTree::Transform& basePose,
+                                                        iDynTree::Twist& baseTwist)
+{
+    iDynTree::Position imuPosition;
+    iDynTree::toEigen(imuPosition) = state.imuPosition;
+    iDynTree::Rotation imuRotation;
+    iDynTree::toEigen(imuRotation) = state.imuOrientation.toRotationMatrix();
+    auto A_H_IMU = iDynTree::Transform(imuRotation, imuPosition);
+
+    iDynTree::Twist v_IMU;
+    iDynTree::Vector3 imuLinVel, imuAngVel;
+    iDynTree::toEigen(imuLinVel) = state.imuLinearVelocity;
+    iDynTree::toEigen(imuAngVel) = state.imuOrientation.toRotationMatrix()*(meas.gyro - state.gyroscopeBias);
+    v_IMU.setLinearVec3(imuLinVel);
+    v_IMU.setAngularVec3(imuAngVel);
+
+    if (!m_modelComp.getBaseStateFromIMUState(A_H_IMU, v_IMU, basePose, baseTwist))
+    {
+        return false;
```

Comment:
Probably you can print something here. It may help you while debugging

Reply:
Done in https://github.com/dic-iit/bipedal-locomotion-framework/commit/415e1729f4d047cfa520ce00633f7ede70796167